storage/aptosdb/src/lib.rs

```diff
@@ -1006,100 +867,278 @@ impl AptosDB {
         Ok(())
     }
 
-    fn commit_ledger_and_state_kv_db(
+    fn calculate_and_commit_ledger_and_state_kv(
         &self,
-        last_version: Version,
-        ledger_schema_batches: LedgerSchemaBatches,
-        sharded_state_kv_batches: ShardedStateKvSchemaBatch,
-        new_root_hash: HashValue,
-        ledger_info_with_sigs: Option<&LedgerInfoWithSignatures>,
+        txns_to_commit: &[impl Borrow<TransactionToCommit> + Sync],
+        first_version: Version,
+        expected_state_db_usage: StateStorageUsage,
+        sharded_state_cache: Option<&ShardedStateCache>,
+    ) -> Result<HashValue> {
+        let new_root_hash = thread::scope(|s| {
+            let _timer = OTHER_TIMERS_SECONDS
+                .with_label_values(&["save_transactions__work"])
+                .start_timer();
+            // TODO(grao): Write progress for each of the following databases, and handle the
+            // inconsistency at the startup time.
```

Comment:
not related to this pr, but we can't recover if it crashes in the middle? or we just ignore anything that's not covered by ledger info?

Reply:
I have a TODO to truncate those versions that are larger than ledger info version. It's probably not that critical if we assume there is no fork.

(It seems there are some places that call get_latest_transaction_info_option to get the latest version, we probably want to get the ledger info version instead, I will work on it in a separate PR)

Reply:
truncate is not a high pri, but what happens if txn has a different version than event for example. I thought state sync is using the last transaction info to fetch instead of last ledger info?

Reply:
Yes, that's a problem I was saying in the "()" above. I need to get it done before 1.6 cut. (it's not in the 1.5)

Reply:
>  get the ledger info version instead

the overall progress, you mean

Reply:
yes ^